Proper handling of received frames.

MarekPorwisz opened the pull request:
Internal review.

Created API to pass received frame to OpenThread thread and used it to pass 15.4 packets.

MarekPorwisz committed:
net: ieee802154: Allow event callback configuration

Implement a way to pass handler for various events from
radio driver

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

MarekPorwisz committed:
drivers: ieee082154_nrf5: Tx started notification

Notify about actual data transmission start.
Needed when ACK timeout is disabled in the radio driver.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

MarekPorwisz committed:
net: openthread: Tx started notification

OpenThread uses "tx started" notification to start ACK
timeout if the driver does not support it.
In order for this to be achieved tx task needs to run in parallel
to the OpenThread task, otherwise it may delay the gap between
cca and tx or run after transmission has already finished.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

mniestroj committed:
net: wifi: fix log message printing ssid and psk

The following logs show up when trying to connect:

  uart:~$ wifi connect xxxxxx xxxxxxxxxx
  Connection requested
  <dbg> net_wifi_mgmt.wifi_connect: (0x20000500): xxxxxx 6 255 1 \
                xxxxxxxxxx 10
  <err> log: argument 6 in source net_wifi_mgmt log message \
                "%s: (%p): %s %u %u %u %s %u" missinglog_strdup().
  <err> log: argument 2 in source net_wifi_mgmt log message \
                "%s: (%p): %s %u %u %u %s %u" missinglog_strdup().
  Connected
  uart:~$

Fix that by using LOG_HEXDUMP_DBG() to print SSID and PSK.

Signed-off-by: Marcin Niestroj <m.niestroj@grinn-global.com>

mbolivar-nordic committed:
soc: nordic_nrf: validate base addresses with new DT API

Use the new devicetree API in a file which cross-checks dt reg base
addresses with values from the vendor HAL.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

MarekPorwisz committed:
net: lib: openthread: Create API for RX frame handling

OpenThread received frame handling shall be done in the OpenThread
thread.
Created API to pass net packets to the proper thread and handle them
there.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

MarekPorwisz committed:
net: openthread: Pass received frame to the proper thread.

Received frames shall be handled in the OpenThread thread, not in
the receiver thread.
Passed received frame to the function that will handle it in a proper
thread instead of calling otPlatRadioReceiveDone directly.

Signed-off-by: Marek Porwisz <marek.porwisz@nordicsemi.no>

mbolivar-nordic committed:
dts: nordic_nrf: add flash_controller labels

Make sure every flash controller has a node label "flash_controller".
This will make it easier to refer to the SoC NVMC node when necessary.

Signed-off-by: Martí Bolívar <marti.bolivar@nordicsemi.no>

rlubos commented on subsys/net/l2/openthread/openthread.c:
```diff
@@ -256,30 +256,15 @@ static enum net_verdict openthread_recv(struct net_if *iface,
-		otPlatRadioReceiveDone(ot_context->instance,
-				       &recv_frame, OT_ERROR_NONE);
+	if (notify_new_rx_frame(pkt) != OT_ERROR_NONE) {
+		NET_ERR("No bufferts for packet.");
```
typo, s/bufferts/buffers

rlubos commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -256,6 +260,40 @@ static inline void handle_tx_done(otInstance *aInstance)
+	net_pkt_unref(pkt);
+}
+
+otError notify_new_rx_frame(struct net_pkt *pkt)
```
It's Zephyr internal API, I think we should not use OT error space here. A typical approach for Zephyr (0 on success, `errno` on failure) would be fine.

rlubos commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -45,7 +45,11 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME, CONFIG_OPENTHREAD_L2_LOG_LEVEL);
+#define OT_RX_MESSAGES 5
+
 enum pending_events {
+
```
Extra empty line.

rlubos commented on subsys/net/lib/openthread/platform/radio.c:
```diff
@@ -256,6 +260,40 @@ static inline void handle_tx_done(otInstance *aInstance)
+{
+	otError err = OT_ERROR_NO_BUFS;
+
+	if (k_msgq_put(&rx_pkt_queue, &pkt, K_NO_WAIT) == 0) {
```
Perhaps we should block here (`K_FOREVER`) instead of dropping the frame? I think that eventually, the lowest layer (radio driver) should start dropping frames in case upper layers do not catch up processing them. Otherwise, we might end-up dropping frames that are considered "delivered" (acknowledged).

rlubos replied:
In order to achieve that though, a fix in the radio shim would be needed as well (block instead of drop):
https://github.com/zephyrproject-rtos/zephyr/blob/6667dc35e7ea75eb19602ffc86c694ed29587b2c/drivers/ieee802154/ieee802154_nrf5.c#L101

ioannisg committed:
arch: arm: aarch32: cortex_m: write 0x0 to reserved exception entries

Write 0x0 instead of z_arm_reserved to vector exception
entries that are always reserved for future use by the
ARM architecture. These vector table entries cannot be
fetched to be executed by the Cortex-M exception entry,
so having z_arm_reserved gives a false impression, since
it is a function that may be invoked in the code. This
modification is safe since these vector entries are also
not supposed to be read / written by the code.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: aarch32: add z_arm_reserved only when we have SysTick

If the Cortex-M core does not implement the SysTick peripheral,
we should not be adding z_arm_reserved in the corresponding
vector table entry. If we do have SysTick implemented but we
are not using it as the system timer, we shall install the
reserved interrupt at the vector table entry.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: aarch32: cortex_m: add z_arm_reserved only if core has SE

If the Cortex-M core does not implement the Security Extension,
we should not be adding z_arm_reserved in the corresponding
vector table entry. That is because the entry is reserved by
the ARM architecture.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ioannisg committed:
arch: arm: aarch32: rename z_arm_reserved to z_arm_exc_spurious

In the Cortex-M exception table we rename z_arm_reserved()
function to z_arm_exc_spurious(), as it is invoked when
existing (that is, non-reserved) but un-installed exceptions
are triggered, accidentaly, by software, or hardware. This
currently applies to SysTick and SecureFault exceptions.

Since fault.S is shared between Cortex-M and other AARCH32
architectures, we keep z_arm_reserved as a defined symbol
there. This commit does some additional, minor, "no-op"
cleanup in #ifdef's for Cortex-M and Cortex-R.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

erwango committed:
drivers: flash: stm32: Use correct driver compatible

Use valid stm32 flash driver compatibles instead of flash
area compatible.
For 'write_block_size' property, use reference to soc-nv-flash.

fixes #23997

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

galak committed:
kconfig: Add dt_node_int_prop_{int,hex}

Add kconfigfunctions that given the property name to an integer type
property will return its value as either an string int or string hex
value.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
drivers: ethernet: sam_gmac: replace dt_int_val in Kconfig

Replace dt_int_val as its deprecated and meant to be removed with
dt_node_int_prop_int.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

ioannisg committed:
tests: arch: arm: interrupt: add test-case for spurious exception

We add a simple test case to verify the behavior
of z_arm_exc_spurious handler. We use the SysTick
interrupt for that so the test is enabled for
platforms that have but do not use the SysTick.

Signed-off-by: Ioannis Glaropoulos <Ioannis.Glaropoulos@nordicsemi.no>

ABOSTM committed:
dts/arm/st: f2 USB OTGFS clock on AHB2 bus

On F2, USB OTGFS in on AHB2 bus.

Signed-off-by: Alexandre Bourdiol <alexandre.bourdiol@st.com>

joerchan committed:
Bluetooth: host: Fix wrong latency and timeout values in conn update

Fix the pending slave set connection latency and timeout values not used
in the connection update procedure when
CONFIG_BT_GAP_PERIPHERAL_PREF_PARAMS has not been enabled.

Signed-off-by: Joakim Andersson <joakim.andersson@nordicsemi.no>

kapi-no committed:
boards: arm: nrf5340_dk_nrf5340: init shared memory to zero on reset

This change ensures that shared memory doesn't hold any pre-reset
values after Soft reset is issued. When the Network Core boots up,
shared memory is always set to zero.

Signed-off-by: Kamil Piszczek <Kamil.Piszczek@nordicsemi.no>

Commit:
drivers: video: Add dedicated video init priority

Create a dedicated config symbol for video device initialization.
Generally, video device init is low priority comparing to standard
devices. Moreover some video device can rely on other device init,
like the csi mcux driver which rely on sensor and i2c init.

This fixes a crash in video capture sample, when camera sensor
(mt9m114) is not connected.

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

Commit:
sample: video: capture: video software generator fallback

Always try to fallback to video software pattern generator, allowing
to run the sample, even when there is no real sensor connected.
(e.g. mimxrt1064 without mt9m114).

Signed-off-by: Loic Poulain <loic.poulain@linaro.org>

carlescufi committed:
doc: redirects: Fix nRF board redirects

Fix the board redirects. The following boards existed before we moved to
a new style of URL for boards, and thus each have 2 entries (old style
old name to old style new name; new style old name to new style
new name):

- nrf51_pca10028
- nrf52810_pca10040
- nrf52840_pca10056
- nrf52840_pca10059
- nrf52_pca10040
- nrf52_pca20020

The following boards did not exist (or were not referenced) before we
moved to the new style of URL for boards, and thus only have one entry
(new style old name to new style new name):

- nrf51_pca10031
- nrf52811_pca10056
- nrf52833_pca10100
- nrf52840_pca10090
- nrf5340_dk_nrf5340
- nrf9160_pca10090

Signed-off-by: Carles Cufi <carles.cufi@nordicsemi.no>

pabigot committed:
samples: drivers: fix ps2 sample for timer api change

Sleep duration is given in integral milliseconds so use k_msleep()
instead of k_sleep().

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

pabigot committed:
drivers: flash: sam: fix timeout units

The duration is used in math with the uptime clock, not as a timeout.
Correct value expression and rename to clarify units.

Signed-off-by: Peter Bigot <peter.bigot@nordicsemi.no>

galak committed:
kconfig: Remove deprecated functions from kconfigfunctions.py

dt_int_val, dt_hex_val, and dt_str_val have been deprecated for two
releases and thus are meant to be removed.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

galak committed:
Bluetooth: controller: specify entropy device explicitly

For the ll code we know the exact entropy device as its the one for the
SoC that we are on.  So use the new DT macro's to get the entropy device
via DT_INST rather than using CONFIG_ENTROPY_NAME.

Signed-off-by: Kumar Gala <kumar.gala@linaro.org>

nashif committed:
sanitycheck: fix coverage capturing

We got the gcov with the coverage tool (lcov vs gcovr) confused..

Fixes #24003

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
ci: do not use latest breathe release for docs

breathe v4.15.0 was just released and fixes some compatibility issues
with latest sphinx, the combo works, however with many warnings that we
still need to either fix or whitelist. This is temporary until we are
able to use those new versions.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>
